index/matchtree.go

```diff
@@ -1435,3 +1457,18 @@ func isRegexpAll(r *syntax.Regexp) bool {
 		return false
 	}
 }
+
+func queryMetaCacheKey(field string, value *regexp.Regexp) string {
+	sum := sha256.Sum256([]byte(fmt.Sprintf("%s:%s", field, value.String())))
+	return fmt.Sprintf("Meta:%x", sum[:])
+}
+
+func queryRepoIdsCacheKey(repos []zoekt.Repository) string {
+	var b strings.Builder
+	for _, r := range repos {
+		b.WriteString(fmt.Sprint(r.ID))
+		b.WriteByte(',')
+	}
+	sum := sha256.Sum256([]byte(b.String()))
```

Comment:
rather than allocating a string, just write directly into a hasher. You can avoid allocating a huge intermediate string then (plus it will be faster)

Reply:
This makes sense. Updated. Thanks!